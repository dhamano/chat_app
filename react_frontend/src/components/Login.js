import React, { useState } from 'react';
import { NavLink } from 'react-router-dom';
import { login, register } from '../services';
import { setLocalStorage } from '../utilities';

const Login = props => {
    const [password, setPassword] = useState(false);
    const [confirmPassword, setConfirmPassword] = useState(false);
    const [passwordErr, setPasswordErr] = useState(false);

    function handleOnChange (e) {
        if(e.target.name === 'username') props.loginRegVals.setUsername(e.target.value);
        if(e.target.name === 'password') setPassword(e.target.value);
        if(e.target.name === 'confirm-password') setConfirmPassword(e.target.value);
    };

    async function handleSubmitLogin (e) {
        e.preventDefault();
        console.log('login', props.loginRegVals.username, password);
        await login({ username: props.loginRegVals.username, password })
                            .then(res => {
                                console.log('login res', res);
                                setLocalStorage("token", res.data.token);
                                setLocalStorage("username", res.data.username);
                                props.history.push('/')
                            })
    }

    async function handleSubmitRegister (e) {
        e.preventDefault();
        if( password !== confirmPassword) {
            setPasswordErr(true);
            return;
        }
        await register({ username: props.loginRegVals.username, password })
                            .then(res => {
                                console.log('register', res);
                                if ( res.status === 201 ) {
                                    setLocalStorage("token", res.data.token);
                                    setLocalStorage("username", res.data.username);
                                    props.history.push('/')
                                } else {
                                    console.log('ERROR');
                                }
                            })
    }

    return (
        <div className="login-reg-page">
            <div className="login-reg-container">
                <div className={`form-container ${props.loginRegVals.loginOrReg === 'Login' ? 'login' : 'register'}`}>
                    <ul>
                        <li><NavLink onClick={props.loginRegVals.onClickHandler} exact to="/">Login</NavLink></li>
                        <li><NavLink onClick={props.loginRegVals.onClickHandler} to="/register">Register</NavLink></li>
                    </ul>
                    { props.loginRegVals.loginOrReg === "Login" ? (
                    <div className="login">
                        <form onSubmit={handleSubmitLogin}>
                            <div>
                                <input id="username" onChange={handleOnChange} value={!props.loginRegVals.username ? '' : props.loginRegVals.username} name="username" type="text" placeholder="username" autoComplete="username" required />
                                <label htmlFor="username">username</label>
                            </div>
                            <div>
                                <input id="password" onChange={handleOnChange} value={!password ? '' : password} name="password" type="password" placeholder="password" autoComplete="current-password" required />
                                <label htmlFor="password">password</label>
                            </div>
                            <button type="submit">Login</button>
                        </form>
                    </div>
                    ) : (
                    <div className="register">
                        <form onSubmit={handleSubmitRegister}>
                            <div>
                                <input id="reg-username" onChange={handleOnChange} value={!props.loginRegVals.username ? '' : props.loginRegVals.username} name="username" type="text" placeholder="username" autoComplete="username" required />
                                <label htmlFor="reg-username">username</label>
                            </div>
                            <div>
                                <input id="reg-password" className={ passwordErr ? 'error' : '' } onChange={handleOnChange} value={!password ? '' : password} name="password" type="password" placeholder="password" autoComplete="new-password" required />
                                <label htmlFor="reg-password">password</label>
                            </div>
                            <div>
                                <input id="confirmPassword" className={ passwordErr ? 'error' : '' } onChange={handleOnChange} value={!confirmPassword ? '' : confirmPassword} name="confirm-password" type="password" placeholder="confirm password" autoComplete="new-password" required />
                                <label htmlFor="confirmPassword">confirm password</label>
                            </div>
                            <button type="submit">Register</button>
                        </form>
                    </div>
                    )}
                </div>
            </div>
        </div>
    )
};

export default Login;