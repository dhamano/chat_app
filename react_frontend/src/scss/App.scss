.not-compatible {
width: 100vw;
height: 100vh;
background-color: rgba(0, 0, 0, 0.9);
text-align: center;
vertical-align: middle;

    h2 {
        width: 100vw;
        height: 100vh;
        margin: 0;
        color: #fff;
        font-size: 2rem;
        line-height: 1.25em;
        vertical-align: middle;
        display: table-cell;
    }

}

.App {
  text-align: center;
}
