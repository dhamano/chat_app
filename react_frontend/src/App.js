import React, { useState, useEffect } from 'react';
import { Redirect, Route } from 'react-router-dom';

import PrivateRoute from './utilities/PrivateRoute';
import Login from './components/Login';
import Messaging from './components/Messaging';

function App() {

    const [isCompatible, setIsCompatible] = useState(true);
    const [userColor, setUserColor] = useState(false);
    const [username, setUsername] = useState(false);
    const [loginOrReg, setLoginOrReg] = useState('Login');

    const [userMessage, setUserMessage] = useState('')

    window.WebSocket = window.WebSocket || window.MozWebSocket;

    useEffect(() => {
        function checkIfCompaitbleWithWebSocket () {
            let compatible = window.WebSocket || window.MozWebSocket;
            if(compatible) return true;
            return false
        }

        let isCompatible = checkIfCompaitbleWithWebSocket();
        setIsCompatible(isCompatible);
        return function cleanup() {
            checkIfCompaitbleWithWebSocket();
        };
    }, []);

    if (!isCompatible) {
        return (
            <div className="not-compatible">
                <h2>Your browser does not support the WebSocket protocol.<br />Please install a browser that does.</h2>
            </div>
        );
    };

    // const connection = new WebSocket('ws://127.0.0.1:8000');

    // connection.onopen = function() {
        
    // }

    function onClickHandler(e) {
        if(e.target.innerText === "Login") setLoginOrReg('Login');
        if(e.target.innerText === "Register") setLoginOrReg('Register');
    }

    let TEMP_REMOVE_WHEN_USED = {
        userColor,
        setUserColor,
        userMessage,
        setUserMessage
    }

    // console.log("TEMP_REMOVE_WHEN_USED",TEMP_REMOVE_WHEN_USED);
    
    let loginRegVals = {
        setUsername,
        username,
        loginOrReg,
        setLoginOrReg,
        onClickHandler
    } 

    return (
        <div className="App">
           <Route exact path="/" render={ props => localStorage.getItem("token") ? <Redirect to="/home" /> : <Login {...props} loginRegVals={loginRegVals}  /> } />
           <Route path="/register" render={ props => <Login {...props} loginRegVals={loginRegVals}  /> } />
           <PrivateRoute path="/home" component={Messaging} />
        </div>
    );
}

export default App;
