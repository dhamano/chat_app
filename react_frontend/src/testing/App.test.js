import React from 'react';
import { renderer } from '@testing-library/react';
import { Router, withRouter, MemoryRouter } from 'react-router-dom';

import { shallow, mount, render } from 'enzyme';

import App from '../App';
import Login from '../components/Login';

// test('renders learn react link', () => {
//     const AppWithRouter = withRouter(App);
//     const { getByText } = render(<Router><AppWithRouter /></Router>);
//     // const linkElement = getByText(/Login/i);
//     // expect(linkElement).toBeInTheDocument();
// });


describe('<App />', () => {
    it('should match snapshot', () => {
        const AppWithRouter = withRouter(App);
        const tree = renderer.create(<Router><AppWithRouter /></Router>).toJSON();
        expect(tree).toMatchSnapshot();
      });

    it('renders a page', () => {
        const wrapper = shallow(<App />);
        console.log("WRAPPER", wrapper);
        expect(wrapper.find(Login)).to.have.lengthOf(0);
        expect(wrapper.find(Login).shallow().find('.login-reg-page').to.have.lengthOf(1));
    })
});