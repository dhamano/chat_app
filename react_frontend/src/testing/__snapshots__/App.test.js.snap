// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should match snapshot 1`] = `
<div
  className="App"
>
  <Route
    exact={true}
    path="/"
    render={[Function]}
  />
  <Route
    path="/register"
    render={[Function]}
  />
  <PrivateRoute
    component={[Function]}
    path="/home"
  />
</div>
`;
